components/apimgt/org.wso2.carbon.apimgt.micro.gateway/org.wso2.carbon.apimgt.micro.gateway.api.synchronizer/src/main/java/org/wso2/carbon/apimgt/micro/gateway/api/synchronizer/APISynchronizer.java

```diff
@@ -0,0 +1,617 @@
+                // time of creating an API
+                String providerDomain =
+                        MultitenantUtils.getTenantDomain(apidto.getProvider());
+                String removeStr = "/t/" + providerDomain;
+                String contextReturned = apidto.getContext();
+                String context = contextReturned.replaceAll(removeStr, APISynchronizationConstants.EMPTY_STRING);
+                apidto.setContext(context);
+                APIMappingUtil.apisUpdate(apidto);
+            } catch (APISynchronizationException e) {
+                log.error("Failed to create API " + apidto.getId());
+            }
+        }
+        log.info("API synchronization completed.");
+    }
+
+    /**
+     * Method to generate an access token to invoke publisher REST API
+     *
+     * @return AccessTokenDTO
+     */
+    private AccessTokenDTO getAccessToken() throws APISynchronizationException {
+        if (log.isDebugEnabled()) {
+            log.debug("Registering client with dynamic client registration endpoint.");
+        }
+        try {
+            // Registering API client and obtaining a response with consumer key and consumer secret values.
+            OAuthApplicationInfoDTO oAuthDto = TokenUtil.registerClient();
+
+            // Generating an access token to invoke publisher REST API
+            String combinedScopes = APISynchronizationConstants.API_VIEW_SCOPE + " "
+                    + APISynchronizationConstants.API_MEDIATION_POLICY_VIEW_SCOPE;
+            if (log.isDebugEnabled()) {
+                log.debug("Generating an access token with scope(s): " + combinedScopes);
+            }
+
+            return TokenUtil.generateAccessToken(oAuthDto.getClientId(),
+                    oAuthDto.getClientSecret(), combinedScopes);
+        } catch (OnPremiseGatewayException e) {
+            throw new APISynchronizationException("Failed to generate an access token.", e);
+        }
+    }
+
+    /**
+     * Method to update APIs
+     */
+    public void updateApis() throws APISynchronizationException {
+        JSONArray updatedApiIds = getIdentifiersOfUpdatedAPIs();
+        if (updatedApiIds.size() != 0) {
+            synchronizeApis(updatedApiIds);
+        }
+    }
+
+    /**
+     * Method to obtain complete details of all APIs
+     *
+     * @param accessTokenDTO access token DTO
+     * @return A list of APIDTO objects
+     */
+    private List<APIDTO> getDetailsOfAllAPIs(AccessTokenDTO accessTokenDTO)
+            throws APISynchronizationException {
+        APIListDTO summarizedApiDTOList = getAPIList(accessTokenDTO);
+        List<APIInfoDTO> apiInfoList = summarizedApiDTOList.getList();
+        List<APIDTO> apiDTOList = new ArrayList<>();
+        for (APIInfoDTO apiInfoObj : apiInfoList) {
+            String id = apiInfoObj.getId();
+            String detailedAPIViewUrl = apiPublisherUrl + APISynchronizationConstants.API_VIEW_PATH +
+                    "/" + id;
```

Comment:
multiple usages of "/". Use File.separator as a constant

Reply:
File.separator cannot be used here since this is a url, not a file path. So if File.separator is used it would not work in a Windows environment.

Reply:
Yeah I missed it. I wanted to point out multiple usages of "/". Then use a constant